test/Microsoft.AspNet.Mvc.Core.Test/ControllerActionInvokerTest.cs

```diff
@@ -2117,14 +2151,17 @@ public class TestControllerActionInvoker : ControllerActionInvoker
                       valueProviderFactoryProvider,
                       actionBindingContext)
             {
-                _factoryMock = controllerFactoryMock;
+                ControllerFactory = controllerFactory;
             }
 
+            public MockControllerFactory ControllerFactory { get; }
+
             public async override Task InvokeAsync()
             {
                 await base.InvokeAsync();
 
-                _factoryMock.Verify();
+                // Make sure that the controller was disposed in every test that creates ones.
+                ControllerFactory.Verify();
```

Comment:
I was very pleasantly surprised to see we already have thorough testing in place for controller disposal. I added some more verification to the tests that shouldn't see the controller created at all.